cdb2jdbc/src/main/java/com/bloomberg/comdb2/jdbc/DatabaseDiscovery.java

```diff
@@ -479,20 +477,18 @@ static void getDbHosts(Comdb2Handle hndl, boolean refresh) throws NoDbHostFoundE
 
             /* if the handle asks for cache, do it now. */
             if (hndl.age != 0) {
-                synchronized(lock) {
-                    TimeAndHosts record = comdb2lcldb.get(
-                            hndl.myDbName + "/" + hndl.myDbCluster);
-                    if (record != null) { /* found */
-                        long currtm = System.currentTimeMillis();
-                        if (currtm <= record.ms) { /* not expired */
-                            hndl.myDbHosts.addAll(record.hosts);
-                            if (debug) { 
-                                System.out.println("Found " + hndl.myDbName + " in hash" 
-                                        + record.hosts + " port is overridden:" +hndl.overriddenPort); 
-                            }
-                            for (int i = 0, len = hndl.myDbHosts.size(); i != len; ++i)
-                                hndl.myDbPorts.add(hndl.overriddenPort);
+                TimeAndHosts record = comdb2lcldb.get(
+                        hndl.myDbName + "/" + hndl.myDbCluster);
+                if (record != null) { /* found */
```

Comment:
As far as I can see, the removal of the lock is safe here:

- No other `hndl` operations are synchronized in this method.
- With `ConcurrentHashMap` `comdb2lcldb.get` is fully safe.
- The `TimeAndHosts` entries are never modified after they are created in `private static void queryDbHosts`.